synapse/handlers/federation.py

```diff
@@ -1711,17 +1710,46 @@ def add_display_name_to_third_party_invite(self, event_dict, event, context):
         defer.returnValue((event, context))
 
     @defer.inlineCallbacks
-    def _validate_keyserver(self, event, auth_events):
-        token = event.content["third_party_invite"]["signed"]["token"]
+    def _check_signature(self, event, auth_events):
```

Comment:
This probably deserves some doc string, just so I can tell whether the behaviour is deliberate or not.